src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -380,4 +382,28 @@ export class NotebookEditor implements INotebookEditor {
                 .ignoreErrors();
         }
     }
+
+    private resetExecutionState() {
+        // After kernel restart, clear timer and check mark for VS Code notebooks only.
+        // This is intended to provide a clearer visual indication that the kernel has
+        // been restarted
+        const editor = this.vscodeNotebook.notebookEditors.find((item) => item.document === this.document);
+        if (editor) {
+            chainWithPendingUpdates(editor.document, (edit) => {
+                const newCells = this.document.getCells().map((cell) => {
+                    const outputs = [...cell.outputs];
+                    return new NotebookCellData(
+                        cell.kind,
+                        cell.document.getText(),
+                        cell.document.languageId,
+                        outputs,
+                        cell.metadata,
+                        // Retain execution order to match Jupyter behavior
+                        { executionOrder: cell.latestExecutionSummary?.executionOrder }
+                    );
+                });
+                edit.replaceNotebookCells(editor.document.uri, new NotebookRange(0, this.document.cellCount), newCells);
```

Comment:
@rebornix @roblourens Is there a better way to achieve this, rather than having to delete & re-create all of the cells.

Reply:
I assume that if it was running previously you have an execution task, can't you use that and end the execution?

Reply:
Oh, you are trying to clear the state after it has finished executing. Do you want to clear outputs too? If so you can run the clear outputs command. Or, you can create execution tasks and complete them immediately with no success or time. Trying to figure out what a better way to do this would look like and I don't really see how it fits into the model. I think that doing it with execution tasks would be reasonable though. Basically that's just how execution-related state is manipulated.

Reply:
If we can think of a general-purpose reason to have this, we could add a "clear execution state" command (like "clear outputs" but doesn't touch outputs)

Reply:
> I think that doing it with execution tasks would be reasonable though. Basically that's just how execution-related state is manipulated.

Yup thats what I thought as well. 
In general I'm assuming thats better than using the replace cells (as that could have other perf implications as well, on the UI & live share, & other areas).

Thanks @roblourens

Reply:
JFYI, we don't want to clear outputs in order to match Jupyter behavior.